dbt_common/helper_types.py

```diff
@@ -36,6 +36,7 @@ class IncludeExclude(dbtClassMixin):
 
     include: Union[str, List[str]]
     exclude: List[str] = field(default_factory=list)
+    silence: List[str] = field(default_factory=list)
```

Comment:
I believe a more appropriate place for this attribute would be on WarnErrorOptions itself, since `IncludeExclude` can be used by other interfaces as well that do not support `silence`. The `silence` functionality feels specific to the `WarnErrorOptions` mechanism itself -- which could override `includes` (and still probably leverage super().includes) to incorporate the business logic of handling `self.silenced`.

Reply:
We can do that. We'd also need to override the `__post_init__` to validate the `silence` options. I'm not a huuuge fan of inheriting a class and then having to override every part of it though. Given these classes are still small it isn't a big deal, but much further I'd start strongly considering disconnecting or collapsing the classes.

Reply:
It also means separating out the test cases, which is a smaller cognitive lift

Reply:
>I'm not a huuuge fan of inheriting a class and then having to override every part of it though.

Me neither, but given there is usage of IncludeExclude beyond the WarnErrorOptions class in dbt-core (e.g. in [UnparsedVersion.columns](https://github.com/dbt-labs/dbt-core/blob/528b95cba8a730963efe343edad1ad19ebe42f7d/core/dbt/contracts/graph/unparsed.py#L154)), I'm nervous this will cause unintended changes to additional components beyond WarnErrorOptions.

Reply:
That makes sense. The risk is low in that:
- if nothing is silenced, everything operates normally
- if instantiating with keyword args, and not specifying `silence`, then `silence` is empty
- if instantiating with ordered args, and only specifying two args, then `silence` is empty

However, you're right regardless. The `silence` option really only makes sense from a logging standpoint, which is the privy of `WarnErrorOptions`, not `IncludeExclude`. I've gone and done the refactor in https://github.com/dbt-labs/dbt-common/pull/112/commits/61f7ff92c0bb562d7941ae5e2be98d390e274b86